js.go

```diff
@@ -2534,7 +2532,7 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 				usrMsg, err = checkMsg(msg, true)
 				if err == nil && usrMsg {
 					msgs = append(msgs, msg)
-				} else if noWait && (err == errNoMessages) && len(msgs) == 0 {
+				} else if noWait && (err == ErrNoMessages) && len(msgs) == 0 {
```

Comment:
Again here we capture. Have you seen a case where we return errNoMessages? I think that it is then a loophole and we should not have this condition. Instead we should have either Timeout or if there is any message returned, the error is cleared (see end of function).